### DISCUSSION TITLE
Can't get dropdown to generate lists for assigning songs on form

### DISCUSSION THREAD

**Discussion by nashvegasnate** [nss-evening-cohort-14/evening-client-side#336] [Capstones!] [answered]
Name: Nathan Keith

Issue: I am trying to use the reactstrap dropdown on AddSongForm and EditSongForm so that the user can assign a particular song to a specific list, however, the form will not show any of the lists. I am using .map to show all the user's lists but nothing is showing up. Below is the current code I have in the AddSongForm:
``` <Label>ASSIGN SONG TO LIST</Label>
          <Input
            type="select"
            name="firebaseKey"
            placeholder="Assign to List"
            id="exampleSelect"
            onChange={handleInputChange}
          >
          {lists?.map((list) => (
            <option key={list.firebaseKey} value={list.firebaseKey}>
              {list.title}
            </option>
          ))}
         </Input> 
```

What I have Googled:
https://stackoverflow.com/questions/44364502/how-to-set-selected-item-in-reactstrap-dropdown
https://github.com/reactstrap/reactstrap/issues/559
https://www.pluralsight.com/guides/how-to-capture-the-value-of-dropdown-lists-with-react-bootstrap

GitHub Links:
https://github.com/nashvegasnate/SetList/blob/development/src/components/Forms/AddSongForm.js
https://github.com/nashvegasnate/SetList/blob/development/src/components/Forms/EditSongForm.js

**Comment by chiestroud** [comment]
Nathan,  I think you commented out 'const [lists, setLists] in Songs.js, but you may need that and pass that to below AddSongForm. 

![image](https://user-images.githubusercontent.com/68720317/123029924-8b008b80-d3a7-11eb-8136-fd1216692a76.png)

And for EditSongForm, if the 'lists' is coming from SongCard, you probably need to pass lists={lists} in SongCard also?

![image](https://user-images.githubusercontent.com/68720317/123030255-22fe7500-d3a8-11eb-8c80-b2dd0a2ecc82.png)

If that's is not what you are trying to do, sorry :D

**Reply by nashvegasnate** [reply]
@chiestroud Thanks so much for the suggestion, however, I couldn't get it to work when I tried it :( Maybe I'm not doing it right?
I am going to keep on keeping on though. Thank you for looking into it!!

**Reply by chiestroud** [reply]
I hope you will get it to work! Another thing I thought about was that you have Lists.js right? And you are mapping 'lists' in the file. So if you copy and paste the above `<input> ~ </input>` code in the Lists.js return statement, then it technically should give you the dropdown list with all the title. If that works, it is giving you what you need. So you can have the same useState & useEffect in other view and as long as you prop drill correctly you should get that working :) Good luck!

**Reply by nashvegasnate** [reply]
Thanks, although I'm not really sure I understand! :-/
And with all the changes I've made in the past couple of days (with the help of others), now I can't delete any songs! Yikes!

**Reply by chiestroud** [reply]
Explaining code is really hard isn't it? lol I was trying to say that you can copy and paste below code in Link.js and the page should give you dropdown list with all the title (it seemed like you have useState and useEffect to get the data and map in Link.js).  If it does give you the correct data in Link.js, you can basically copy and paste the same useState and useEffect in other views to have the dropdown list working :) 
`   <Input
            type="select"
            name="firebaseKey"
            placeholder="Assign to List"
            id="exampleSelect"
            onChange={handleInputChange}
          >
          {lists?.map((list) => (
            <option key={list.firebaseKey} value={list.firebaseKey}>
              {list.title}
            </option>
          ))}
         </Input> `

**Comment by drteresavasquez** [comment, marked as answer]
@nashvegasnate do you have a flow chart created? If so, you need to either update it or reference it. Where you call [SongCard](https://github.com/nashvegasnate/SetList/blob/78ca9db7ee4ddd79e6010795d02ebcf489126401/src/views/SingleList.js#L29)  and define it [here](https://github.com/nashvegasnate/SetList/blob/78ca9db7ee4ddd79e6010795d02ebcf489126401/src/components/Cards/SongCard.js#L16) the card is expecting a lists prop but you never pass it one. 

You also need to go through and make all of your props required so that you can debug this stuff yourself. You should know what components need what props and where they are coming from as a part of your capstone.

**Reply by nashvegasnate** [reply]
I tried adding the lists prop, however, it didn't seem to work. I will go through and mark all props as "required" for debugging.

**Reply by nashvegasnate** [reply]
So, I moved my List state to the index.js in order to have access to lists and songs and added lists={lists} and setLists={setLists} to my routes and the dropdown worked! 
Now, the issue that remains is specific to my SingleList view. The songs which were assigned to particular lists will not show up in the appropriate SingleList view for the selected list. Also, while in the SingleList view, the information on the song card will disappear when deleted, however, the body of the card still remains.

**Reply by drteresavasquez** [reply]
Sounds like you have a lot of work to do. You really need to take a step back and try and understand what you are trying to accomplish. We cannot help you with every step of the application. This is your capstone, so work through how to get to your MVP. I never saw a flow chat. Does it exist and is it updated and correct?

**Reply by drteresavasquez** [reply]
Also, that second part sounds like a new discussion ticket with all the details of what you tried, etc.

**Reply by nashvegasnate** [reply]
I had submitted two prior flow charts and am just now updating flow chart 3.0. Yes, I have a lot of work to do and it is unclear if MVP will be met on time. A new ticket is soon to come forth on the most recent issues,
